### PR TITLE
CanonicalIdentifier in Events (plus some cleanups)

### PR DESCRIPTION
**Pull request by konradkonrad**
This adds the `CanonicalIdentifier` for #3493 to the classes in `raiden/transfer/events.py` (where applicable).
It also adds a transient `.canonical_identifier` property to `NettingChannelState` (based on the already existing partial identifiers), which allows for a lot of code simplification.
`pack_reward_proof` now also uses a `CanonicalIdentifier` as an argument (instead of the singular arguments).
This is still all done without serialization changes.

### CODE REVIEW

File: raiden/transfer/events.py [hunk 1]
```diff
@@ -171,15 +176,15 @@ class ContractSendChannelUpdateTransfer(ContractSendExpirableEvent):
     def __init__(
             self,
             expiration: BlockExpiration,
-            channel_identifier: ChannelID,
-            token_network_identifier: TokenNetworkID,
+            # FIXME: balance_proof below needs to be fully identified, so we could get rid of this
```
Comment (jomuel) [inline raiden/transfer/events.py]: Is it not? It contains all three fields of the canonical identifier already.

Comment (rakanalh) [reply raiden/transfer/events.py]: I am not sure i understand what is meant by the comment... could you please clarify this a bit more?

Comment (jomuel) [reply raiden/transfer/events.py]: It was a reply to the fixme comment. The balance proof contains all information of the canonical identifier already, so the canonical_identifier argument is redundant.

Comment (konradkonrad) [reply raiden/transfer/events.py]: I realized the FIXME comment was misleading...I changed it and use `balance_proof.canonical_identifier` inside the constructor.

**Pull request merged**